library/timing.c

```diff
@@ -399,7 +399,7 @@ int mbedtls_timing_self_test(int verbose)
     int hardfail = 0;
     struct mbedtls_timing_hr_time hires;
     uint32_t a = 0, b = 0;
-    mbedtls_timing_delay_context ctx;
+    mbedtls_timing_delay_context ctx = {};
```

Comment:
To pass the build with clang without warnings, you need to pass all the elements of a structure. `= {0}` is valid C, but Clang warns about it unless there's a single scalar in the structure. [`= {}` is not valid C99](https://en.cppreference.com/w/c/language/struct_initialization#Notes).

`memset` to all-bits-zero would be a correct initialization here. (Throughout the library, we assume that initializing a structure to all-bits-zero is correct even if that includes some padding.) It avoids having to care about the number of fields.

For consistency with the rest of the library, we should probably provide an API function `mbedtls_timing_delay_init` to do the initialization. And also a corresponding `mbedtls_timing_delay_free` function to zero out the structure; since there is no confidential data and no embedded structure, the `free` function would just do `memset` to 0, exactly like the `init` function.

Reply:
I did the memset.
Please note that on current development branch this code has been removed, so maybe we don't want to add new APIs just for this error? In any case, I would like to avoid to write new APIs, because I'm not confident enough with the code, I'm just trying to fix build failures on the current one if possible